contracts/exchange-forwarder/contracts/src/MixinExchangeWrapper.sol

```diff
@@ -139,31 +184,24 @@ contract MixinExchangeWrapper is
         bytes[] memory signatures
     )
         internal
-        returns (FillResults memory totalFillResults)
+        returns (
+            FillResults memory totalFillResults,
+            uint256 wethSpentAmount,
+            uint256 makerAssetAcquiredAmount
+        )
     {
-        bytes memory makerAssetData = orders[0].makerAssetData;
-        bytes memory wethAssetData = WETH_ASSET_DATA;
-
         uint256 ordersLength = orders.length;
-        uint256 makerAssetFilledAmount = 0;
         for (uint256 i = 0; i != ordersLength; i++) {
+            require(
```

Comment:
See above comment ^. You can use the same RichError to replace both of these `require` statements